module.groovy

```diff
@@ -151,6 +151,30 @@ def createModule(String name) {
 }
 
 /**
+ * Update a given module.
+ * @param name the name of the module to update
+ */
+def updateModule(String name) {
+    println "Attempting to update module $name"
+    File targetDir = new File("modules/$name")
+    if (!targetDir.exists()) {
+        println "Module \"$name\" not found"
+        return
+    }
+
+    def moduleGit = Grgit.open(dir: targetDir)
+    def clean = moduleGit.status().clean
+    println "Is \"$name\" clean? $clean"
+    if (!clean) {
+        println "Module has uncommitted changes. Aborting."
+        return
+    }
+
+    println "Updating module $name"
+    moduleGit.pull remote: "origin", branch: "master"
```

Comment:
A thought here: it isn't guaranteed that the user has the `master` branch checked out. Plenty of modules will work out of the `develop` branch and I would think if we hard coded `master` and a user is on `develop` they'll end up pulling from a branch they haven't got checked out.

IIRC you can use `git pull` without a branch argument and that'll simply pull whatever you have checked out. Maybe there is a variant like that we can use instead? Try to test it out  - what happens if the user has `develop` checked out and how can we better support that? :-)

Reply:
Oh I thought most modules are developed directly on master. Will change it.